### PR TITLE
docs: add dynamic routes page

### PR DESCRIPTION
**Pull request by Valerioageno**

### CODE REVIEW

**Review by marcalexiei — DISMISSED** [review]
I created a separate commit with some improvements.
Let me know what you think about them!

File: apps/documentation/src/routes/documentation/routing/dynamic-routes.mdx [hunk 1]
```diff
@@ -13,19 +13,25 @@ import Breadcrumbs, { Element } from '@/components/breadcrumbs'
...
 
 ## Convention
 
-A Dynamic Segment can be created by wrapping a
-file or folder name in square brackets: `[segmentName]`. For example, `[id]` or `[slug]`.
+A dynamic segment can be created by wrapping a
+file or a folder name in square brackets: `[segmentName]`.
 
-## Example
+You are going to probably use something like:
```
Comment (Valerioageno) [inline apps/documentation/src/routes/documentation/routing/dynamic-routes.mdx:24]: I'm not a fan of "personalizing" the documentation by pointing directly the reader. 
I'll think about a rephrase over this sentence

File: apps/documentation/src/routes/documentation/routing/dynamic-routes.mdx [hunk 2]
```diff
@@ -21,10 +21,14 @@ you can use dynamic segments, which are populated at request time.
 A dynamic segment can be created by wrapping a
 file or a folder name in square brackets: `[segmentName]`.
 
-You are going to probably use something like:
+The dynamic segment can be used in both folder and file name:
```
Comment (marcalexiei) [inline apps/documentation/src/routes/documentation/routing/dynamic-routes.mdx:24]:
```suggestion
The dynamic segment can be used in folders and files as well:
```

File: apps/documentation/src/routes/documentation/routing/dynamic-routes.mdx [hunk 3]
```diff
@@ -21,10 +21,14 @@ you can use dynamic segments, which are populated at request time.
...
 
-- `[id]`
-- `[slug]`
+- `src/routes/blog/[slug].tsx`
+- `src/routes/blog/[slug]/index.tsx`
+
+The same path can have multiple dynamic segments:
```
Comment (marcalexiei) [inline apps/documentation/src/routes/documentation/routing/dynamic-routes.mdx:29]:
```suggestion
A path can contain multiple dynamic segments, allowing for more flexible routing.
```

**Review by marcalexiei — CHANGES_REQUESTED** [review]

**Pull request merged**